src/api/dashboard.js

```diff
@@ -56,14 +63,29 @@ Dashboard.prototype = {
     this._dashboard.vis.mapvis.map.setBounds([state.map.ne, state.map.sw]);
   },
 
-  onStateChanged: function (callback, shareURLs) {
-    this._dashboard.vis.once('dataviewsFetched', function () {
-      this._dashboard.widgets._widgetsCollection.initialState();
-      this._dashboard.widgets._widgetsCollection.each(function (m) {
-        m.applyInitialState();
-      }, this);
-      shareURLs === true && this._bindChange(callback);
-    }, this);
+  _widgetsAlreadyInitialized: function () {
+    var widgetsCollection = this._dashboard.widgets._widgetsCollection;
```

Comment:
If we could avoid this kind of 🚋 🚋 🚋  it'd be 🔝 . We're breaking the Law of Demeter & we're accessing private attributes, which might change in the future and break this...  We could just ask `this_dashboard` if it has widgets and they are initialised: 

```
if (this._dashboard.areWidgetsInitialised()) {
 // Do wathever
}
```

I know `this._dashboard` is not a "real" object so we might need to convert it into one.